Common/util/string_utils.cpp

```diff
@@ -203,5 +204,12 @@ void StrUtil::WriteString(const String &s, Stream *out)
         out->Write(s.GetCStr(), len);
 }
 
+void StrUtil::WriteCStr(const String &s, Stream *out, int max_len)
+{
+    max_len = Math::Min(max_len, s.GetLength());
```

Comment:
I can make it return actually written length probably?

This function is meant to forcefully truncate string, it emulates how AGS was writing limited fields originally, so it is not like an erroneous situation.